include/seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp

```diff
@@ -0,0 +1,404 @@
+ * returned type over the underlying matrix.
+ *
+ *  * `value_type`: The derived type must define a `value_type` type which determines the reference type of the
+ *                  inner range.
+ *  * `column_data_view_type`: A type that models std::ranges::RandomAccessRange and std::ranges::View and is the
+ *                            internal type used to iterate over a matrix column.
+ *  * `make_proxy`: This function is called by the dereferencing an iterator over the inner column range and must
+ *                  return a type that is convertible to the reference type of the column (see `value_type`).
+ *  * `initialise_column`: This function is called when incrementing the outer matrix iterator over the columns. It is
+ *                         used to initialise the column view stored inside of the column type to alias the corresponding
+ *                         memory in the matrix' data storage.
+ *  * `on_column_iterator_creation`: This function is called when calling begin on the inner column type.
+ *  * `before_column_iterator_increment`: This function is called before incrementing the internal column iterator.
+ *  * `after_column_iterator_increment`: This function is called after incrementing the internal column iterator.
+ */
+template <typename derived_t>
+class alignment_matrix_column_major_range
+{
+private:
+
+    //!\brief Befriend the derived type.
+    friend derived_t;
+
+    /*!\brief Represents a column within an alignment matrix.
+     * \implements std::ranges::View
+     * \implements std::ranges::InputRange
+     *
+     * \details
+     *
+     * The alignment column depends on the `column_data_view_type` of the derived type. This type is used to
+     * alias the actual data storage of the underlying matrix implementation. such that it does not own the data.
+     */
+    class alignment_column_type : public std::ranges::view_interface<alignment_column_type>
+    {
+    private:
+
+        //!\brief A view aliasing the actual stored data column within the underlying matrix.
+        using view_type = typename deferred_type<typename derived_t::column_data_view_type>::type;
+
+        static_assert(std::ranges::RandomAccessRange<view_type>, "Column view must support random access.");
+        static_assert(std::ranges::View<view_type>, "Column view must be a view.");
+
+        //!\brief The sentinel type of the underlying view.
+        using sentinel = std::ranges::sentinel_t<view_type>;
+
+        /*!\brief The iterator over an alignment column.
+         *
+         * \details
+         *
+         * The iterator represents the current cell within an alignment column. When dereferenced it
+         * calls the function `make_proxy` of the derived class to obtain a proxy value over the current matrix cell.
+         * This proxy type depends on the derived type.
+         */
+        class iterator_type
```

Comment:
No, because they are strongly coupled. Making them external classes will produce another set of issues since the correct types and data must be available. Since friends cannot be transferred it will get messy to provide the correct information.

Reply:
Okay, I see the point. It already helps a lot that you marked the end of the classes with respective comments.